h2o-core/src/main/java/water/H2O.java

```diff
@@ -2298,4 +2286,31 @@ public static H2ONode getClientByIPPort(String ipPort){
   public static Key<DecryptionTool> defaultDecryptionTool() {
     return H2O.ARGS.decrypt_tool != null ? Key.<DecryptionTool>make(H2O.ARGS.decrypt_tool) : null;
   }
+
+  public static URI downloadLogs(URI destinationDir, LogArchiveContainer logContainer) {
+    return downloadLogs(destinationDir.toString(), logContainer);
+  }
+
+  public static URI downloadLogs(URI destinationDir, String logContainer) {
+    return downloadLogs(destinationDir, LogArchiveContainer.valueOf(logContainer));
+  }
+
+  public static URI downloadLogs(String destinationDir, LogArchiveContainer logContainer) {
+    String outputFileStem = LogsHandler.getOutputLogStem();
+    String outputFileName = outputFileStem + "." + logContainer.getFileExtension();
+    byte[] logBytes = LogsHandler.downloadLogs(logContainer, outputFileStem);
+    File destination = new File(destinationDir, outputFileName);
+
+    try (FileOutputStream fileOutputStream = new FileOutputStream(destination)) {
+      fileOutputStream.write(logBytes);
+    } catch (IOException e) {
+      e.printStackTrace();
```

Comment:
Shouldn't the reaction be more complex, e.g. use our `Log` class ?

Reply:
Agree. The main purpose of this change was refactoring, I would like to avoid mixing fixes into this change. But I can improve this in the following PR, what do you think?

Reply:
Sure.